server/server.go

```diff
@@ -86,9 +88,12 @@ func New(log logrus.FieldLogger, config Config, api APIDefinition) (*Server, err
 }
 
 func (s *Server) Shutdown(to time.Duration) error {
+	s.doneOnce.Do(func() {
+		defer close(s.done)
+	})
```

Comment:
the defer is called once the function it is in ends. which in this case is immediately, so you can leave off the keywork. the concern of catching panics is not given here.

Reply:
Ah yes, good catch. I think we still want to defer the closing of the channel though. Updated.